java/rocksjni/rocksjni.cc

```diff
@@ -1587,3 +1587,74 @@ jlong Java_org_rocksdb_RocksDB_getUpdatesSince(JNIEnv* env,
   rocksdb::RocksDBExceptionJni::ThrowNew(env, s);
   return 0;
 }
+
+//////////////////////////////////////////////////////////////////////////////
+// rocksdb::DB::AddFile
+
+void add_file_helper(JNIEnv* env, jobjectArray& jfile_path_list,
+                     std::vector<std::string>& file_path_list) {
+  const int jfile_path_list_len =
+      static_cast<int>(env->GetArrayLength(jfile_path_list));
+  for (int i = 0; i < jfile_path_list_len; i++) {
+    jstring jfile_path =
+        static_cast<jstring>(env->GetObjectArrayElement(jfile_path_list, i));
+    const char* file_path = env->GetStringUTFChars(jfile_path, NULL);
+    file_path_list.push_back(std::string(file_path));
+    env->ReleaseStringUTFChars(jfile_path, file_path);
+  }
+}
+
+void add_file_helper(
+    JNIEnv* env, jlongArray jfi_handle_list,
+    std::vector<rocksdb::ExternalSstFileInfo>& file_info_list) {
+  const int len = static_cast<int>(env->GetArrayLength(jfi_handle_list));
```

Comment:
You could consider passing the size of the `jlongArray` as a `jint` from Java. This pattern is used elsewhere and I suspect it is cheaper than calling `GetArrayLength`